### PR TITLE
docs(core): add ! for viewchild

### PR DESCRIPTION
**Pull request by behroozbc**
fix ! for ts2564 error

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #42811

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by behroozbc** [comment]
@googlebot I signed it!

File: packages/core/src/metadata/di.ts [hunk 1]
```diff
@@ -338,7 +338,7 @@ export interface ViewChildrenDecorator {
    * with `@ViewChildren('cmp')`)
    *   * Any provider defined in the child component tree of the current component (e.g.
    * `@ViewChildren(SomeService) someService: SomeService`)
-   *   * Any provider defined through a string token (e.g. `@ViewChildren('someToken') someTokenVal:
+   *   * Any provider defined through a string token (e.g. `@ViewChildren('someToken') someTokenVal!:
```
Comment (gkalpak) [inline packages/core/src/metadata/di.ts:341]: :+1:
Can you please update other examples that need this (for example, the one above and the one below this one) :pray:

**Review by gkalpak — COMMENTED** [review]
Good catch! Thx, @behroozbc :+1:
The fix is not ideal, but I don't think we have better options anyway 😅

**Review by gkalpak — COMMENTED** [review]
BTW, could you also update the commit message to more closely follow our [commit message guidelines](https://github.com/angular/angular/blob/655507012b7b59c83e27429349ce3ac10608a82d/CONTRIBUTING.md#commit). For example, include a reference to the associated issue so it is closed once the PR is merged:

> docs(core): ...
>
> ...
>
> Fixes #42811

File: packages/core/src/metadata/di.ts (hunk 1 above)
Comment (behroozbc) [reply packages/core/src/metadata/di.ts:341]: yes

**Review by behroozbc — COMMENTED** [review]
> BTW, could you also update the commit message to more closely follow our [commit message guidelines](https://github.com/angular/angular/blob/655507012b7b59c83e27429349ce3ac10608a82d/CONTRIBUTING.md#commit). For example, include a references to the associated issue so it is closed once the PR is merged:
> 
> > docs(core): ...
> > ...
> > Fixes #42811

okay I will fix it 
thx for helping me

File: packages/core/src/metadata/di.ts (hunk 1 above)
Comment (gkalpak) [reply packages/core/src/metadata/di.ts:341]: Thx!

**Comment by gkalpak** [comment]
Superceded by #42816.

**Pull request closed without merging**